carla_controller/simulation.py

```diff
@@ -22,35 +60,42 @@ def __init__(self):
         self.vehicle_blueprints = [blueprint for blueprint in self.vehicle_blueprints if blueprint.id not in BANNED_VEHICLES]
 
         self._recording = False
+        if output_directory is None:
+            output_directory = DEFAULT_OUTPUT_DIRECTORY
+        self.output_directory = output_directory
+        # Ensure that the directory is created and ready to receive data
+        
+        Path(f"{self.output_directory}/{self.run_id}/rgb").mkdir(parents=True, exist_ok=True)
+        Path(f"{self.output_directory}/{self.run_id}/semantic").mkdir(parents=True, exist_ok=True)
 
         self.actors = []
+        self.cameras = []
+
+        self.default_vehicles = vehicles
+        self.default_pedestrians = pedestrians
+        self.default_pedestrian_cross_percentage = pedestrian_cross_percentage
+        self.default_hero_spawn_point = hero_spawn_point
+
 
     def spawn_hero_vehicle(self, spawn_point = None):
```

Comment:
a "hero vehicle"  Haha, is that a unique CARLA term or just something for fun.

Reply:
I've seen it in other literature via CARLA so I'm assuming it's used there, but it's at least clear what it does. I originally wrote a version of this where each car, or N cars, get cameras, but a lot of processing problems popped up as soon as I spread it out so I reduced us back down to one car.